src/main/scala/com.snowplowanalytics.iglu/core/ParseError.scala

```diff
@@ -15,27 +15,33 @@ package com.snowplowanalytics.iglu.core
 /** Common error type for parsing core Iglu entities */
 sealed trait ParseError extends Product with Serializable {
   def code: String
+  def message(data: String): String
 }
 
 object ParseError {
   case object InvalidSchemaVer extends ParseError {
     def code = "INVALID_SCHEMAVER"
+    def message(data: String) = s"Invalid schema version $data, code: $code"
```

Comment:
The problem with this detailed message is that it will be added to every failure (read bad row) and it will be too much information for something that repeats so often. It's up to user-facing assets to give detailed info and how to fix a problem. But assets where this method is used are usually tend to provide semi-human-readable messages (like bad rows).